code/modules/mob/living/simple_animal/borer/borer.dm

```diff
@@ -0,0 +1,799 @@
+				chemicals++
+			else if(chemicals < 250)
+				chemicals+=2
+			if (chemicals > 250)
+				chemicals = 250 //to prevent 251 chemical bug from +2 per tick
+
+
+		if(stat != DEAD && victim.stat != DEAD)
+
+			if(victim.reagents.has_reagent("sugar"))
+				if(!docile)
+					if(controlling)
+						victim << "<span class='warning'>You feel the soporific flow of sugar in your host's blood, lulling you into docility.</span>"
+					else
+						src << "<span class='warning'>You feel the soporific flow of sugar in your host's blood, lulling you into docility.</span>"
+					docile = 1
+			else
+				if(docile)
+					if(controlling)
+						victim << "<span class='warning'>You shake off your lethargy as the sugar leaves your host's blood.</span>"
+					else
+						src << "<span class='warning'>You shake off your lethargy as the sugar leaves your host's blood.</span>"
+					docile = 0
+			if(controlling)
+
+				if(docile)
+					victim << "<span class='warning'>You are feeling far too docile to continue controlling your host...</span>"
+					victim.release_control()
+					return
+
+				if(prob(5))
+					victim.adjustBrainLoss(rand(1,2))
+
+				if(prob(victim.brainloss/20))
+					victim.say("*[pick(list("blink","blink_r","choke","aflap","drool","twitch","twitch_s","gasp"))]")
+
+/mob/living/simple_animal/borer/say(message)
+	if(dd_hasprefix(message, ";"))
+		message = copytext(message,2)
+		for(var/borer in borers)
+			borer << "<span class='borer'>Cortical Link: [truename] sings, \"[message]\""
+		for(var/mob/dead in dead_mob_list)
+			dead << "<span class='borer'>Cortical Link: [truename] sings, \"[message]\""
+		return
+	if(!victim)
+		src << "<span class='warning'>You cannot speak without a host!</span>"
+		return
+	if(message == "")
+		return
+
+/mob/living/simple_animal/borer/UnarmedAttack(mob/living/M)
+	healthscan(usr, M)
+	chemscan(usr, M)
+	return
+
+/mob/living/simple_animal/borer/ex_act()
+	if(victim)
+		return
+
+	..()
+
+/mob/living/simple_animal/borer/proc/Infect(mob/living/carbon/victim)
+	if(!victim)
+		return
+
+	if(victim.borer)
+		src << "<span class='warning'>[victim] is already infested!</span>"
+		return
+
+	if(!victim.key || !victim.mind)
+		src << "<span class='warning'>[victim]'s mind seems unresponsive. Try someone else!</span>"
+		return
+
+	if (victim && victim.dna && istype(victim.dna.species, /datum/species/skeleton))
+		src << "<span class='warning'>[victim] does not possess the vital systems needed to support us.</span>"
+		return
+
+	src.victim = victim
+	victim.borer = src
+	forceMove(victim)
+
+	log_game("[src]/([src.ckey]) has infested [victim]/([victim.ckey]")
+
+/mob/living/simple_animal/borer/proc/leave_victim()
+	if(!victim)
+		return
+
+	if(controlling)
+		detatch()
+
+	src.forceMove(get_turf(victim))
+
+	victim.borer = null
+	reset_perspective(null)
+
+	var/mob/living/V = victim
+	V.verbs -= /mob/living/proc/borer_comm
+	victim = null
+	return
+
+/mob/living/simple_animal/borer/verb/infect_victim()
```

Comment:
I've always been wondering as to why. Why are action buttons preferred to verbs?